modules/promo-code/promo-code.permissions.js

```diff
@@ -1,18 +1,32 @@
-const { allow } = require('graphql-shield');
-const { hasRoles } = require('../../utils/rules');
+const { allow, and } = require('graphql-shield');
+
+const {
+  promoCodeInputValidator,
+} = require('../../validators/promocode-validator');
+const { hasRoles, inputDataValidation } = require('../../utils/rules');
 const {
   roles: { ADMIN, SUPERADMIN },
 } = require('../../consts');
+const {
+  PROMOCODE_FIELDS: { PROMOCODE },
+} = require('../../consts/promocode-fields');
 
 const promoCodeQuery = {
+  getPromoCodeById: allow,
```

Comment:
everybody can do it? even logged out users?

Reply:
Improve permission